### PR TITLE
Remove dbus cpp library

### PR DESCRIPTION
**Pull request by al1img**
Remove third-party dbus c++ lib and use glib high API to handel DBus server.

### CODE REVIEW

File: src/Config.hpp [hunk 1]
```diff
@@ -15,248 +15,227 @@
  *   along with this program; if not, write to the Free Software
  *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
  *
- * Copyright (C) 2016 EPAM Systems Inc.
+ * Copyright (C) 2017 EPAM Systems Inc.
```
Comment (andr2000) [inline src/Config.hpp:18]: 2017?

File: src/DBusControl.h [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+ * it was derived from.
+ */
+
+#ifndef __DBUSCONTROL_H__
+#define __DBUSCONTROL_H__
+
+#include <gio/gio.h>
+
+G_BEGIN_DECLS
+
```
Comment (andr2000) [inline src/DBusControl.h:14]: extra new line?

File: src/DBusControl.h [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+/* ------------------------------------------------------------------------ */
+/* Declarations for com.epam.DisplayManager.Control */
+
+#define TYPE_CONTROL (control_get_type ())
```
Comment (andr2000) [inline src/DBusControl.h:19]: why space after name?

File: src/DBusControl.h [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+struct _ControlIface
+{
+  GTypeInterface parent_iface;
+
+  gboolean (*handle_user_event) (
+    Control *object,
+    GDBusMethodInvocation *invocation,
+    guint arg_event);
+
```
Comment (andr2000) [inline src/DBusControl.h:36]: extra line?

File: src/DBusControl.h [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+};
+
+GType control_get_type (void) G_GNUC_CONST;
+
+GDBusInterfaceInfo *control_interface_info (void);
```
Comment (andr2000) [inline src/DBusControl.h:41]: camelCase?

File: src/DBusControl.h [hunk 6]
```diff
@@ -0,0 +1,179 @@
+/*
+ * Generated by gdbus-codegen 2.56.4. DO NOT EDIT.
```
Comment (andr2000) [inline src/DBusControl.h]: why do we need a generated file here?

File: src/Config.hpp (hunk 1 above)
Comment (al1img) [reply src/Config.hpp:18]: This header was copied from displ_be with wrong year. The real real of display manager creating is 2017.

File: src/DBusControl.h (hunk 6 above)
Comment (al1img) [reply src/DBusControl.h]: done

File: README.md [hunk 7]
```diff
@@ -328,4 +328,18 @@ Example:
...
 ```
 DisplayManager -c display.cfg -v *:Debug
+```
+
+## Send user commands with DBus:
+
+Getting introspect:
+
+```bash
+dbus-send --session --print-reply --dest=com.epam.DisplayManager /com/epam/DisplayManager org.freedesktop.DBus.Introspectable.Introspect
```
Comment (andr2000) [inline README.md]: s/session/system

File: README.md [hunk 8]
```diff
@@ -328,4 +328,18 @@ Example:
...
+```
+
+Sending user event `1`:
+
+```bash
+dbus-send --session --dest=com.epam.DisplayManager --type=method_call  /com/epam/DisplayManager com.epam.DisplayManager.Control.userEvent uint32:1
```
Comment (andr2000) [inline README.md]: ditto

**Comment by andr2000** [comment]
With README fixed with system/session parameter:
`Reviewed-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>`
`Tested-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>`

**Pull request merged**